### PR TITLE
refactor: remove `namespace` keyword

### PR DESCRIPTION
**Pull request by jaschdoc**
Removes `namespace` keyword and `/` separator.

- [x] Add Weeder check ensuring namespaces / modules begin with uppercase symbol

Depends on https://github.com/flix/flix/pull/5687
Related to https://github.com/flix/flix/issues/4800

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala [hunk 1]
```diff
@@ -1830,20 +1830,11 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
...
     }
 
-    /**
-      * Namespaces are lower or uppercase.
-      */
-    // TODO NS-REFACTOR remove
-    def Namespace: Rule1[Name.NName] = rule {
-      SP ~ oneOrMore(UpperCaseName).separatedBy("/" | ".") ~ SP ~>
-        ((sp1: SourcePosition, parts: Seq[Name.Ident], sp2: SourcePosition) => Name.NName(sp1, parts.toList, sp2))
-    }
```
Comment (jaschdoc) [inline main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1840]: Should this also be entirely removed? The comment on ns-refactor wants it removed, but the `Names.DotSeparated` rule allows for both `UppercaseName` and `LowerCaseName` so they are not entirely equivalent. Maybe @mlutze can comment on what is okay here?

Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1840]: I think we should have something in `Weeder` to check that the names in a namespace are all uppercase.

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1840]: As in, we have it, or we should have it?

Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:1840]: Should

**Comment by magnus-madsen** [comment]
@JonathanStarup @mlutze @stephentetley This PR changes `namespace` to `mod` and the separate from `/` to `.`

I will not merge it yet, but just a heads up. You can already use `mod` and `.` in the current version of Flix, if you want to do an incremental refactor.

**Comment by magnus-madsen** [comment]
@jaschdoc PR looks strange now...

**Comment by jaschdoc** [comment]
> @jaschdoc PR looks strange now...

I implemented a check in the Weeder to ensure that namespaces are formatted properly as suggested by Matt in https://github.com/flix/flix/pull/5692#discussion_r1158326565

**Comment by magnus-madsen** [comment]
Everyone- I would like to merge this soon. Could I ask you to update your projects? :)

**Pull request merged**